### PR TITLE
 	new file:   blog/_posts/2020-01-15-affiliations-feature.md

### PR DESCRIPTION
**Pull request by seasidesparrow**
new file:   blog/images/blog_2020-01-15_example.png
 	new file:   blog/images/blog_2020-01-15_facet.png
 	new file:   blog/images/blog_2020-01-15_matched.png
 	new file:   blog/images/blog_2020-01-15_unmatched.png

### CODE REVIEW

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+We've recently changed the way we index affiliations by introducing a new search field: "affil".  Affil combines all of the available affiliation data -- raw strings, canonical strings, IDs, and abbreviations into a single, searchable field.  It's intended to be a more comprehensive search of both raw strings and the enriched institution asssignments we've applied to our data.  So for example, searching for [affil:"UCB"] (https://ui.adsabs.harvard.edu/search/fl=identifier&q=affil:"UCB") will return papers where 'UCB' matches some part of the raw affiliation, but *also* return affiliations we've matched to the University of California at Berkeley.  Because "UCB" is use by some authors to abbreviate other universities (for example "Catholic University of Brasilia, Brazil" or "University of Colorado, Boulder"), the affil search also finds these, but they can be de-selected in the affiliation facet.
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Fixed both.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Users are reminded that while affiliation information is largely complete for recent refereed literature, not all records contains an affiliation, therefore, searching by affiliation alone will inherently be incomplete.  We strongly recommend combining affiliation searches with author searches for best results.
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Fixed.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+So, you could search for [aff:Harvard](https://ui.adsabs.harvard.edu/search/fl=identifier&q=aff:"Harvard") and get back all affiliations that contain Harvard in the affiliation string - including "Harvard Street".  To ensure you get only the University, you could search for [aff:"Harvard University"](https://ui.adsabs.harvard.edu/search/fl=identifier&q=aff:"Harvard%20University") but that would return only affiliation strings with that exact phrase, excluding "Harvard Univ.", "Harvard U", etc.  Better would be to use the identifier for Harvard University by searching [aff_id:A00211](https://ui.adsabs.harvard.edu/search/fl=identifier&q=aff_id:A00211).  This returns all affiliations which contain Harvard University at the parent level. Best yet would be to search by institution, [inst:"Harvard U"](https://ui.adsabs.harvard.edu/search/fl=identifier&q=inst:"Harvard U")  which returns affiliations which contain Harvard University at the parent level, plus all affiliations for all of Harvard University's children.
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Fixed multiple instances throughout.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### How we got here: the curation and pipeline process
+
+**The human element: curation**
+
+Production of the initial affiliation database began with two human-generated projects: creating a set of institutional identifiers, and matching affiliation metadata to these identifiers as precisely as possible.  Both of these were done almost entirely by Carolyn Grant.
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Done.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
... and then assigning this string its corresponding alpha-numeric ID.  The process sounds straightforward, but extraction of all strings relating to "Department", "Physics", "University of California", and "Irvine" can be long and tedious, especially if you're searching thousands of possible strings.  There are ways to speed that process (e.g. by searching for "Irvine" and "Physics" in all strings, and then looking through just that set of results), but at the beginning it was an entirely human-curated process.  Complicating things is the fact that many authors are affiliated with more than one institution, for example "Astronomy Dept., U. Texas -- Austin, and UNAM-Morelia".  Publishers -- especially smaller publications and conference proceedings -- rarely use standardized text for affiliations, and even more rarely list multiple affiliations in separate entities in their metadata.  So the curation process also involves splitting those multiple affiliations and then characterizing each affiliation separately.  That process was repeated over and over for the 6600 affiliations that each occasionally had thousands or tens of thousands of different non-unique publisher-provided strings.
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Fixed this and a few other instances throughout.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Cleaning the data turned out to be more of an art than a science.  There exist curation tools for cleaning up messy data - the one we used almost exclusively was Open Refine (called Google Refine at the time).  This allowed bulk substitutions, expansions, translations and more.  But cleaning is not enough.  For example, changing all instances of "UC Irvine" to "University of California Irvine" may seem like a good idea, except that new instances of "UC Irvine" are likely to continue coming in.
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Fixed this and a few other instances throughout.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+However, we're getting close to the limit of what we can do with human classification without lots of effort.  Our most recent pass through our metadata has just over 4.6 million unclassified strings.  There are about 6000 unclassified affiliations that appear 100 times or more in the metadata and assigning them aff_ids would provide a lot of new information.  However, over 95 percent of unclassified affiliations occur fewer than 10 times, and **46 percent of affiliation strings only appear once in our affiliation metadata**.
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Modified text to use current numbers (multiple statistics changed throughout).

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+<div class="text-center">
+        <img class="img-thumbnail" src="{{ site.baseurl }}/blog/images/blog_2020-01-15_matched.png" width="50%" /><img class="img-thumbnail" src="{{ site.baseurl }}/blog/images/blog_2020-01-15_unmatched.png" width="50%"/>
+<em>Left: The vast majority of matched publisher-provided affiliation strings in our database appear at least 10 times. Right: XXX</em>
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Done.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+</div>
+<br>
+
+This is too much to work through by hand, but it represents valid affiliation data for millions of author-affiliation pairs. For the unmatched affiliations that occur fewer than 100 times, we're using machine learning techniques to try and assign ids.  For now, we're using scikit-learn tools to try and match unknown strings to IDs. Specifically, we're using scikit-learn's feature_extraction tools for analyzing the known affiliation strings, and the SGDClassifier to generate models.  We then pass the unknown strings through the model and extract both a best match and a confidence estimate.  The process can be very memory intensive because of the large number of classes -- nearly 6600, one for each aff_id.  A typical data set of 25,000 unmatched strings takes about an hour on a modest workstation (Apple iMac with an Intel i5 processor and 24 GB of memory).
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Fixed.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+There are hundreds of thousands of organizations around the world, and in principle any institution or business can be assigned an identifier.  The overwhelming majority of these are not relevant to ADS, so the first task was to establish a list of institutions that match affiliations in our metadata.   The current list stands at about 6600 departments, organizations, and parent institutions in all geographic regions of the world.  Many of these institutions are also tagged with a second identifier, namely the identifier of the parent institution.  For example, the parent institution of the Department of Physics at the University of California at Irvine is "University of California at Irvine".  For now, our affiliations have two levels only, so in this example, we do not assign "The University of California System" as the parent of UC Irvine, but we do have the System identifier in cases of ambiguity (e.g. Univ. of Cal. with no city or other unambiguous identifying information)
```
[the remark this answers is not in the document]
Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: Modified.

File: blog/_posts/2020-01-15-affiliations-feature.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+However, we're getting close to the limit of what we can do with human classification without lots of effort.  Our most recent pass through our metadata has just over 4.6 million unclassified strings.  There are about 6000 unclassified affiliations that appear 100 times or more in the metadata and assigning them aff_ids would provide a lot of new information.  However, over 65 percent of unclassified affiliations occur fewer than 10 times, and **13 percent of unmatched affiliation strings only appear once in all of our affiliation metadata**.
```
Comment (kelockhart) [inline blog/_posts/2020-01-15-affiliations-feature.md]: "...4.6 million unclassified strings" - but the top of the right panel of the figure below says 16.6 million unmatched affiliation strings?

Comment (seasidesparrow) [reply blog/_posts/2020-01-15-affiliations-feature.md]: I clarified this -- it's 4.8 million *unique* strings  which yields a total of 16.6 million total unmapped affiliations.

**Pull request merged**